felt/src/bigint_felt.rs

```diff
@@ -820,25 +820,67 @@ impl fmt::Display for ParseFeltError {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use proptest::prelude::*;
 
     #[test]
-    fn add_felts_within_field() {
-        let a = FeltBigInt::<FIELD_HIGH, FIELD_LOW>::new(1);
-        let b = FeltBigInt::new(2);
-        let c = FeltBigInt::new(3);
+    // Tests that the result of adding two zeros results in zero.
+    fn add_zeros() {
+        let a = FeltBigInt::<FIELD_HIGH, FIELD_LOW>::new(0);
+        let b = FeltBigInt::new(0);
+        let c = FeltBigInt::new(0);
 
         assert_eq!(a + b, c);
     }
 
     #[test]
-    fn add_assign_felts_within_field() {
-        let mut a = FeltBigInt::<FIELD_HIGH, FIELD_LOW>::new(1i32);
-        let b = FeltBigInt::new(2i32);
+    // Tests that the result of performing add asign with two zeros results in zero.
+    fn add_assign_zeros() {
+        let mut a = FeltBigInt::<FIELD_HIGH, FIELD_LOW>::new(0);
+        let b = FeltBigInt::new(0);
         a += b;
-        let c = FeltBigInt::new(3i32);
+        let c = FeltBigInt::new(0);
 
         assert_eq!(a, c);
     }
+    #[test]
+    // Tests that the result of performing a bitwise "and" operation with two zeros results in zero.
+    fn bit_and_zeros() {
+        let a = FeltBigInt::<FIELD_HIGH, FIELD_LOW>::new(0);
+        let b = FeltBigInt::new(0);
+        let c = FeltBigInt::new(0);
+
+        assert_eq!(&a & &b, c);
+    }
+    #[test]
+    // Tests that the result of performing a bitwise "or" operation with two zeros results in zero.
+
+    fn bit_or_zeros() {
+        let a = FeltBigInt::<FIELD_HIGH, FIELD_LOW>::new(0);
+        let b = FeltBigInt::new(0);
+        let c = FeltBigInt::new(0);
+
+        assert_eq!(&a | &b, c);
+    }
+
+    #[test]
+    // Tests that the result of performing a bitwise "xor" operation with two zeros results in zero.
+    fn bit_xor_zeros() {
+        let a = FeltBigInt::<FIELD_HIGH, FIELD_LOW>::new(0);
+        let b = FeltBigInt::new(0);
+        let c = FeltBigInt::new(0);
+
+        assert_eq!(&a ^ &b, c);
+    }
+
+    #[test]
+    // Tests that the result of performing a division between two zeros results in zero.
+    fn div_zeros() {
+        let a = FeltBigInt::<FIELD_HIGH, FIELD_LOW>::new(0);
+        let b = FeltBigInt::new(0);
+        let c = FeltBigInt::new(0);
+
+        assert_eq!(&a / &b, c);
```

Comment:
There is a bug in the test and in the operation, dividing by 0 should panic

Reply:
fixed